arviz/plots/ppcplot.py

```diff
@@ -157,11 +160,10 @@ def plot_ppc(
 
         >>> az.plot_ppc(data, num_pp_samples=30, random_seed=7)
     """
-    for group in ("posterior_predictive", "observed_data"):
-        if not hasattr(data, group):
-            raise TypeError(
-                '`data` argument must have the group "{group}" for ppcplot'.format(group=group)
-            )
+    if group not in ("posterior_predictive", "prior_predictive", "observed_data"):
+        raise TypeError(
+            '`group` argument must be `posterior_predictive`, `prior_predictive` or `observed_data`'
+        )
```

Comment:
I would divide the argument checks in 2 parts. First check that the `group` is one of the valid options, and then make sure that the 2 needed groups are present in the inference data object (like it was done before with the hasattr). Also, note that if `group="posterior"` then the inference data must have posterior predictive and observed data groups but _prior predictive is not needed_

Reply:
Also, make sure to not use `group` as argument and loop variable at the same time ;)

Reply:
Yeah actually I was confused earlier about how to check for the group and display error, now I got it. Thanks :)